src/search-bar.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { updateSearch, updateSearchResults } from './actions';
+import { getAutocomplete } from './fetch-data';
+
+let SearchBarDumb = ({ dispatch, searchInput, searchAutocomplete }) =>
+    <div>
+        <input name="search-box" type="text" className="searchTerm" placeholder="What are you looking for?"
+            value={searchInput} onChange={(event) => {
+                dispatch(updateSearch(event.target.value));
+                if (event.target.value) {
+                    getAutocomplete(event.target.value)
+                    .then(results => dispatch(updateSearchResults(results)));
+                } else {
+                    dispatch(updateSearchResults([]));
+                }
+            }} />
+        <button type="submit" className="searchButton">
+            <i className="fa fa-search"></i>
+        </button>
+        <ul className="setListDropdownMenu">
+            {
+                searchAutocomplete.map(result =>
+                    <li>{result}</li>
+                )
+            }
+        </ul>
+    </div>
+
+let mapStateToProps = (state) => {
+    return { searchInput: state.searchInput,
+             searchAutocomplete: state.searchAutocomplete };
+}
+
+let mapDispatchToProps = (dispatch) => {
+    return { dispatch: dispatch };
+}
+
+let SearchBar = connect(
+    mapStateToProps,
+    mapDispatchToProps
+)(SearchBarDumb);
+
+export default SearchBar;
```

Comment:
Did you mean to move the dropdown  menu to this component?  is it showing up in two places?  

Also, we've had the discussion that my preference is to have a mixture of redux store/component state vs storing everything in redux - in this example, I don't think it is necessary for the store to have access to the value changing in the search bar because no other component is going to need it.  But it works.

Reply:
Ok, gotcha.
I think either one works; my line of thought was that only one component (search bar) needs to know when its state changes and so could store it on its own, versus making it accessible to all components via the redux store.

Reply:
I'll approve the changes now!